CMakeLists.txt

```diff
@@ -100,6 +100,43 @@ set(CMAKE_AUTOUIC
     ON
 )
 
+if (PROJECT_IS_TOP_LEVEL)
```

Comment:
`#TIL` PROJECT_IS_TOP_LEVEL. very nice.

per cmake docs, that was added in cmake 3.21

and we have:
```
cmake_minimum_required(
  VERSION 3.24
)
```

so this is great!